core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/mercury/v03/request.go

```diff
@@ -154,26 +157,31 @@ func (c *client) multiFeedsRequest(ctx context.Context, ch chan<- mercury.Mercur
 			case http.StatusUnauthorized:
 				retryable = false
 				state = encoding.UpkeepNotAuthorized
+				errCode = encoding.HttpToErrCode(resp.StatusCode)
 				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely this is caused by unauthorized upkeep", sl.Time.String(), sl.UpkeepId.String(), resp.StatusCode)
 			case http.StatusBadRequest:
 				retryable = false
 				state = encoding.InvalidMercuryRequest
+				errCode = encoding.HttpToErrCode(resp.StatusCode)
 				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely this is caused by invalid format of timestamp", sl.Time.String(), sl.UpkeepId.String(), resp.StatusCode)
 			case http.StatusInternalServerError, http.StatusBadGateway, http.StatusServiceUnavailable, http.StatusGatewayTimeout:
 				retryable = true
 				state = encoding.MercuryFlakyFailure
+				errCode = encoding.HttpToErrCode(resp.StatusCode)
 				return fmt.Errorf("%d", resp.StatusCode)
 			case http.StatusPartialContent:
 				// TODO (AUTO-5044): handle response code 206 entirely with errors field parsing
 				c.lggr.Warnf("at timestamp %s upkeep %s requested [%s] feeds but mercury v0.3 server returned 206 status, treating it as 404 and retrying", sl.Time.String(), sl.UpkeepId.String(), sl.Feeds)
 				retryable = true
 				state = encoding.MercuryFlakyFailure
+				errCode = encoding.HttpToErrCode(resp.StatusCode)
 				return fmt.Errorf("%d", http.StatusPartialContent)
 			case http.StatusOK:
 				// continue
 			default:
 				retryable = false
 				state = encoding.InvalidMercuryRequest
+				errCode = encoding.ErrCodeBadRequest
```

Comment:
as of current implementation, retryInterval will need to be -1 for this error code to be handled